twilight-lavalink/src/client.rs

```diff
@@ -200,6 +200,7 @@ impl Lavalink {
 
                     if e.channel_id.is_none() {
                         self.sessions.remove(&guild_id);
+                        self.server_updates.remove(&guild_id);
```

Comment:
This is the fix where it removes the old server even if we don't get an update that the server was removed. If we update the voice then we can assume that the server connection isn't valid now.